SignalMessaging/utils/ThreadUtil.m

```diff
@@ -615,11 +618,46 @@ + (ThreadDynamicInteractions *)ensureDynamicInteractionsForThread:(TSThread *)th
                     indicator.timestampForSorting);
             }
         }
+
+        // Determine the position of the focus message _after_ performing any mutations
+        // around dynamic interactions.
+        if (focusMessageId != nil) {
+            result.focusMessagePosition =
+                [self focusMessagePositionForThread:thread transaction:transaction focusMessageId:focusMessageId];
+        }
     }];
 
     return result;
 }
 
+
++ (nullable NSNumber *)focusMessagePositionForThread:(TSThread *)thread
+                                         transaction:(YapDatabaseReadWriteTransaction *)transaction
+                                      focusMessageId:(NSString *)focusMessageId
+{
+    OWSAssert(thread);
+    OWSAssert(transaction);
+    OWSAssert(focusMessageId);
+
+    // Enumerate in reverse to count the number of messages after the "focus message".
+    __block NSUInteger count = 0;
+    __block BOOL didMatch = NO;
+    [[transaction ext:TSMessageDatabaseViewExtensionName]
```

Comment:
This simply enumerates. We could do binary search, but is it worth it?  

* Searches are rare.
* Search results with recent results will be more useful.
* We only enumerate the collections keys, not objects, which will help considerably.

Reply:
The `ensureDynamicInteractions` method is already a (the?) major cost of delay when presenting the ConversationVC, so I'd like to try to avoid introducing a linear scan here, even though it's only when searching.

Any reason we can't use the constant time:

```
- (BOOL)getGroup:(NSString * _Nullable * _Nullable)groupPtr
           index:(nullable NSUInteger *)indexPtr
          forKey:(NSString *)key
    inCollection:(nullable NSString *)collection;
```